src/utils.jl

```diff
@@ -152,72 +160,75 @@ function separate_possible_terms_and_calculate_parameter_constant(
-    for term in terms
-        if (
-            is_variable_in_model(model, term.variable_1) &&
-            is_variable_in_model(model, term.variable_2)
-        )
-            push!(quad_vars, term) # if there are only variables, it remains a quadratic term
-
-        elseif (
-            is_parameter_in_model(model, term.variable_1) &&
-            is_variable_in_model(model, term.variable_2)
-        )
-            push!(quad_aff_vars, term)
-            push!(variables_associated_to_parameters, term.variable_2)
-            push!(model.multiplicative_parameters, term.variable_1.value)
-            push!(
-                aff_terms,
-                MOI.ScalarAffineTerm(
-                    term.coefficient * model.parameters[term.variable_1],
-                    term.variable_2,
-                ),
+    if function_quadratic_terms_has_parameters(model, terms)
+        # When we have a parameter x variable or a variable x parameter the convention is to rewrite
+        # the SQT with parameter as variable_index_1 and variable as variable_index_2
+        for term in terms
+            if (
+                is_variable_in_model(model, term.variable_1) &&
+                is_variable_in_model(model, term.variable_2)
             )
+                push!(quad_vars, term) # if there are only variables, it remains a quadratic term
 
-        elseif (
-            is_variable_in_model(model, term.variable_1) &&
-            is_parameter_in_model(model, term.variable_2)
-        )
-            # Check convention defined above
-            push!(
-                quad_aff_vars,
-                MOI.ScalarQuadraticTerm(
-                    term.coefficient,
-                    term.variable_2,
-                    term.variable_1,
-                ),
+            elseif (
+                is_parameter_in_model(model, term.variable_1) &&
+                is_variable_in_model(model, term.variable_2)
             )
-            push!(variables_associated_to_parameters, term.variable_2)
-            push!(
-                aff_terms,
-                MOI.ScalarAffineTerm(
-                    term.coefficient * model.parameters[term.variable_2],
-                    term.variable_1,
-                ),
+                push!(quad_aff_vars, term)
+                push!(variables_associated_to_parameters, term.variable_2)
+                push!(model.multiplicative_parameters, term.variable_1.value)
+                push!(
+                    aff_terms,
+                    MOI.ScalarAffineTerm(
+                        term.coefficient * model.parameters[term.variable_1],
+                        term.variable_2,
+                    ),
+                )
+
+            elseif (
+                is_variable_in_model(model, term.variable_1) &&
+                is_parameter_in_model(model, term.variable_2)
             )
-            push!(model.multiplicative_parameters, term.variable_2.value)
+                # Check convention defined above
```

Comment:
this convention is used later so that we can know which is a variable and which is a param when re-using the stored info without having to re check which is which.
might be good to add this comment, so people know the reason.